File: Java/TransportService/src/main/java/transport/services/ITransportServices.java
package transport.services;

import transport.model.Flight;
import transport.model.User;

import java.time.LocalDate;
import java.util.List;

public interface ITransportServices {
    void login(String username, String password, ITransportObserver client) throws TransportException;
    void logout(String username, ITransportObserver client) throws TransportException;
    String hashPassword(String password);
    User findUser(String username);
    void addUser(User user);
    boolean buyTicket(String Flight, int seats, List<String> passagers);
    List<String> searchFlights(String destination, LocalDate departure) throws TransportException ;
    List<Flight> getAllFlights() throws TransportException ;

}

File: C#/TransportServices/transport/services/ITransportServices.cs
using TransportModel.transport.model;

namespace TransportServices.transport.services;

public interface ITransportServices {
    bool Login(string username, string password, ITransportObserver client);
    bool BuyTicket(string destination, int requiredSeats, int seats, List<string> passengers);
    List<Flight> SearchFlights(string destination, DateTime departure);
    List<Flight> GetAllFlights();
}